src/Microsoft.Health.Dicom.Functions/Utils/BatchUtils.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Model;
+
+namespace Microsoft.Health.Dicom.Functions.Utils;
+internal static class BatchUtils
```

Comment:
I think it makes sense to centralize this. Although this class sort of looks like the it has methods similar to `Task.Run` so perhaps we should call this `TaskBatch` and the method `TaskBatch.Start`?

I think we can probably make this method more generic like:
```csharp
// Maybe -1 indicates that we run all of the tasks at once?
public static async Task Run<T>(IReadOnlyList<T> input, Func<T, Task> taskFactory, int maxThreads = -1)
```